app/settings/account/-components/connected-emails/component.ts

```diff
@@ -121,34 +136,35 @@ export default class ConnectedEmails extends Component {
         return this.toast.success(successMessage);
     });
 
+    @task({ withTestWaiter: true })
+    onSave = task(function *(this: ConnectedEmails) {
+        let newEmail;
+        try {
+            this.changeset.validate();
+            if (this.changeset.get('isValid') && this.changeset.get('emailAddress')) {
+                this.set('lastUserEmail', this.changeset.get('emailAddress'));
+                newEmail = this.store.createRecord('user-email', {
+                    emailAddress: this.changeset.get('emailAddress'),
+                    user: this.currentUser.user,
+                });
+                yield newEmail.save();
+                this.set('showAddModal', true);
+                this.reloadUnconfirmedList();
+                this.toast.success(this.intl.t('settings.account.connected_emails.save_success'));
```

Comment:
Should probably clear the changeset value after successfully saving the email, otherwise the input box would still have the added email in it.